### PR TITLE
Update Agent to 1.16.1

### PR DESCRIPTION
**Pull request by richardpen**
### Testing

- [ ] Builds on Linux (`make release`)
- [ ] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [ ] Unit tests on Linux (`make test`) pass
- [ ] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [ ] Integration tests on Linux (`make run-integ-tests`) pass
- [ ] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [ ] Functional tests on Linux (`make run-functional-tests`) pass
- [ ] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

This contribution is under the terms of the Apache 2.0 License:

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-1.16.0
\ No newline at end of file
+1.16.1
```
Comment (nmeyerhans) [inline VERSION]: It looks like you introduced a trailing newline here. Did you use `scripts/update-version` to update the version?

Comment (richardpen) [reply VERSION]: I didn't know that we have a script for this, just updated.

**Comment by richardpen** [comment]
@samuelkarp Updated

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
 # Changelog
 
-## Unreleased
-* Bug - Fixed a bug where the agent could miss sending an ENI attachment to ECS because of address propagation delays [#1148](https://github.com/aws/amazon-ecs-agent/pull/1148)
+## 1.16.1
+* Bug - Fixed a bug where the agent could miss sending an ENI attachment to ECS because of address propagation delays. [#1148](https://github.com/aws/amazon-ecs-agent/pull/1148)
```
Comment (samuelkarp) [inline CHANGELOG.md]: Can you wrap these at 80 characters?

Comment (richardpen) [reply CHANGELOG.md]: done

**Comment by richardpen** [comment]
This was merged into master branch.

**Pull request closed without merging**